### PR TITLE
Hybrid map tree display

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/components/network-modification-tree-pane.js [hunk 1]
```diff
@@ -87,4 +93,8 @@ export default NetworkModificationTreePane;
 
 NetworkModificationTreePane.propTypes = {
     studyUuid: PropTypes.string.isRequired,
+    drawerNodeEditorOpen: PropTypes.bool.isRequired,
```
Comment (niconoir) [inline src/components/network-modification-tree-pane.js:96]: Maybe remove these props and store the drawerNodeEditorOpen prop in redux store instead.

File: src/components/network-modification-tree.js [hunk 2]
```diff
@@ -214,3 +332,11 @@ const NetworkModificationTree = ({ treeModel, openModificationDrawer }) => {
 };
 
 export default NetworkModificationTree;
+
+NetworkModificationTree.propTypes = {
+    treeModel: PropTypes.object.isRequired,
+    drawerNodeEditorOpen: PropTypes.bool.isRequired,
```
Comment (niconoir) [inline src/components/network-modification-tree.js:338]: See previous comment about removing drawerNodeEditorOpen prop and storing it to redux store.

File: src/components/network-modification-tree.js [hunk 3]
```diff
@@ -167,12 +204,74 @@ const NetworkModificationTree = ({ treeModel, openModificationDrawer }) => {
...
+                });
+            }
+        }
+    }, [studyMapTreeDisplay, drawerNodeEditorOpen]);
+
     return (
         <>
-            <Box style={styles.container} display="flex" flexDirection="row">
+            <Box
+                style={{ width: '100%', height: '100%' }}
```
Comment (niconoir) [inline src/components/network-modification-tree.js:268]: Use className={classes.container} here

**Review by niconoir — CHANGES_REQUESTED** [review]

File: src/components/network-modification-tree.js (hunk 2 above)
Comment (etiennehomer) [reply src/components/network-modification-tree.js:338]: @AbdelHedhili What do you think about it ? We can discuss it tomorrow

File: src/components/map-lateral-drawers.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            duration: theme.transitions.duration.enteringScreen,
+        }),
+        marginLeft: -drawerExplorerWidth,
+    },
+    nodeEditorShift: {
+        transition: theme.transitions.create('margin', {
+            easing: theme.transitions.easing.easeOut,
+            duration: theme.transitions.duration.enteringScreen,
+        }),
+        pointerEvents: 'none',
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/map-lateral-drawers.js:50]: We need this property to catch the mouse events in the nodeEditor

File: src/components/horizontal-toobar.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                flexDirection: 'row',
+            }}
+        >
+            {!(studyDisplayMode === StudyDisplayMode.TREE) && (
+                <>
+                    <Tooltip
+                        title={intl.formatMessage({ id: 'SubstationList' })}
+                        placement="right"
+                        arrow
+                        enterDelay={1000}
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/horizontal-toobar.js]: done

File: src/components/network-modification-tree.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/network-modification-tree.js]: done

File: src/components/network-modification-tree.js [hunk 6]
```diff
@@ -191,14 +290,33 @@ const NetworkModificationTree = ({ treeModel, openModificationDrawer }) => {
                         snapGrid={snapGrid}
                     >
                         <Controls
-                            style={styles.controls}
+                            style={{
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/network-modification-tree.js]: done

File: src/components/network-modification-tree-pane.js (hunk 1 above)
Comment (etiennehomer) [reply src/components/network-modification-tree-pane.js]: Done

File: src/components/network-modification-tree.js (hunk 2 above)
Comment (etiennehomer) [reply src/components/network-modification-tree.js]: done

File: src/components/map-lateral-drawers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/map-lateral-drawers.js:51]: Ok

File: src/components/study-pane.js [hunk 7]
```diff
@@ -144,9 +144,9 @@ const StudyPane = ({
 
     const classes = useStyles();
 
-    const [drawerExplorerOpen, setDrawerExplorerOpen] = useState(true);
+    // const [drawerExplorerOpen, setDrawerExplorerOpen] = useState(true);
 
-    const [drawerNodeEditorOpen, setDrawerNodeEditorOpen] = useState(false);
+    // const [drawerNodeEditorOpen, setDrawerNodeEditorOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/study-pane.js]: done

File: src/components/network-modification-tree.js [hunk 8]
```diff
@@ -92,10 +90,22 @@ const NetworkModificationTree = ({
 
     const classes = useStyles();
 
+    const isModificationsDrawerOpen = useSelector(
+        (state) => state.isModificationsDrawerOpen
+    );
+
+    const closeModificationsDrawer = () => {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/network-modification-tree.js]: It was removed

**Review by niconoir — APPROVED** [review]
Tested and works fine.

File: src/components/horizontal-toobar.js [hunk 9]
```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
```
Comment (AbdelHedhili) [inline src/components/horizontal-toobar.js:2]: 2022

File: src/components/map-lateral-drawers.js [hunk 10]
```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
```
Comment (AbdelHedhili) [inline src/components/map-lateral-drawers.js:2]: 2022

**Review by AbdelHedhili — APPROVED** [review]
change license header to 2022 where you moved or created a file

File: src/components/map-lateral-drawers.js (hunk 10 above)
Comment (etiennehomer) [reply src/components/map-lateral-drawers.js]: done

**Pull request merged**